File: leetcode/algorithms/557_reverse_words_in_a_string_3/main.py
class ReverseWordsInAStringIII:
    def reverseWords(self, s: str) -> str:
        sArray = s.split(' ')
        for i in range(len(sArray)):
            sArray[i] = sArray[i][::-1]
        return " ".join(sArray)


    # Solution
    # Solution 1
    def solution1(self, s: str) -> str:
        return ' '.join(s.split()[::-1])[::-1]

    # Solution 2
    def solution2(self, s: str) -> str:
        return ' '.join(x[::-1] for x in s.split())

File: leetcode/algorithms/557_reverse_words_in_a_string_3/main.dart
class ReverseWordsInAStringIII {
  String reverseWords(String s) {
    List<String> sArray = s.split(" ");

    for (int i = 0; i < sArray.length; i++) {
      sArray[i] = sArray[i].split("").reversed.join();
    }

    return sArray.join(" ");
  }

  // Solution
  String solution(String s) {
    return s.split(' ').map((e) => e.split('').reversed.join()).join(' ');
  }
}
